ext/openssl/ossl_ssl.c

```diff
@@ -1958,7 +1958,6 @@ ossl_ssl_read_internal(int argc, VALUE *argv, VALUE self, int nonblock)
 	else
 	    rb_str_modify_expand(str, ilen - RSTRING_LEN(str));
     }
-    rb_str_set_len(str, 0);
     if (ilen == 0)
 	return str;
```

Comment:
`str` needs to be cleared here, when trying to read zero bytes (ilen == 0)

Reply:
Ah good catch. This wasn't called by the test because `OpenSSL::Buffering` just returns early if `size == 0`.

But I fixed it in case it's sometimes used without the buffering.